### PR TITLE
chore: update drone format to 1.0

### LINKED ISSUE #6463: Drone 1.0.0 yaml
Now that the Gitea org uses Drone 1.0.0 we should update the .drone.yml to align with the new format. The current one works, however due to the dropping of `groups` it takes longer to run.

I'm thinking we also have separate pipelines for each DB that gets tested, so that the services could be terminated sooner (which would decrease resource usage). If we do this we will probably have to have a build step in each pipeline though (a simple `TAGS="sqlite" make build` would likely be enough). Pipelines supports the `depends_on` param.

We also have an ARM server that can run docker builds to be able to publish arm builds of our docker image.

cc: @sapk (due to your ARM docker efforts)

### PR DESCRIPTION
**Pull request by appleboy**
remove git clone steps and execute the following command to change the format:

```sh
$ drone convert --save
```

fix #6463

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -1,453 +1,571 @@
...
-  git_push:
-    image: appleboy/drone-git-push
-    pull: true
-    secrets: [ git_push_ssh_key ]
-    remote: git@github.com:go-gitea/gitea.git
-    force: false
+    ignore_branch: true
+    project_identifier: gitea
+  environment:
+    CROWDIN_KEY:
```
Comment (tboerger) [inline .drone.yml]: You should convert this kind of environment variables into settings.

File: .drone.yml [hunk 2]
```diff
@@ -1,453 +1,571 @@
...
+    - push
+
+- name: pre-build
+  pull: always
+  image: webhippie/nodejs:latest
+  commands:
+  - npm install
+  - make stylesheets-check
+  when:
+    event:
```
Comment (tboerger) [inline .drone.yml]: Maybe just drop the events entirely?

File: .drone.yml [hunk 3]
```diff
@@ -1,453 +1,571 @@
...
-    when:
-      event: [ push, pull_request ]
-      branch: [ master ]
-
-  release-test:
-    image: golang:1.12
-    pull: true
+  environment:
+    TAGS: bindata sqlite sqlite_unlock_notify
+  when:
```
Comment (tboerger) [inline .drone.yml]: These conditions won't work anymore as far as I know. You should use `ref`

File: .drone.yml [hunk 4]
```diff
@@ -1,453 +1,571 @@
...
-
-pipeline:
-  download_translations:
-    image: jonasfranz/crowdin
-    pull: true
-    secrets: [ crowdin_key ]
-    project_identifier: gitea
-    ignore_branch: true
+steps:
+- name: download_translations
```
Comment (techknowlogick) [inline .drone.yml]: Could you split out translation steps into a different pipeline?

**Comment by kolaente** [comment]
@appleboy How's the status here?

**Comment by appleboy** [comment]
@kolaente I will update asap

**Comment by kolaente** [comment]
@appleboy anything we can help you with?

**Comment by kolaente** [comment]
@appleboy If it's ok for you, I'm going to take over this pr since we need it.

**Comment by appleboy** [comment]
@kolaente OK, Thank you.

**Comment by appleboy** [comment]
@kolaente maybe you can update this PR not create new PR.

**Comment by kolaente** [comment]
@appleboy sure, but I don't have push rights on your branch :sweat_smile:

**Comment by appleboy** [comment]
@kolaente I have invited you as collaborators.

**Comment by kolaente** [comment]
IMHO it may be a better idea to close this pr and continue the work in #7331 since that one is almost finished... but I can also just push my changes to this pr and close the other one.

**Comment by kolaente** [comment]
### Things I've changed:

I've refactored the drone file to use the new syntax of drone 1, which brings several advanteges:
* multiple pipelines
* dependencies of steps -> paralle execution of tests

Multiple pipelines:
* `testing`: Runs all linters, unit and integration tests
* `translations`: Downloads the newest translations from crowdin and also pushes them back to the repo
* `release`: cross-compiles releases with xgo and publishes them
* `docs`: publishes the docs
* `docker`: builds and publishes the docker images
* `notify`: triggers the discord notify hook

From the parallelization the whole pipeline now takes ~15min instead of 35min for a pr! :tada:

File: .drone.yml [hunk 5]
```diff
@@ -1,75 +1,106 @@
...
+        - tag
+        - pull_request
 
-  pre-build:
+steps:
+  - name: fetch-tags
+    pull: default
+    image: docker:git
+    commands:
+      - git fetch --tags --force
```
Comment (techknowlogick) [inline .drone.yml:51]: Can you exclude this step from running on PRs?

File: .drone.yml [hunk 6]
```diff
@@ -1,75 +1,106 @@
...
     when:
-      event: [ push, tag, pull_request ]
+      event:
+        - push
+        - tag
+        - pull_request
 
-  build-without-gcc:
-    image: golang:1.10 # this step is kept as the lowest version of golang that we support
-    pull: true
```
Comment (techknowlogick) [inline .drone.yml:62]: Can you keep comments about this step (as why we use a specific go version and why we build with no tags)

File: .drone.yml (hunk 5 above)
Comment (kolaente) [reply .drone.yml:51]: Done.

File: .drone.yml (hunk 6 above)
Comment (kolaente) [reply .drone.yml:62]: Done.

**Review by techknowlogick — APPROVED** [review]

**Comment by kolaente** [comment]
@tboerger @lunny @lafriks @appleboy please review.

File: .drone.yml [hunk 7]
```diff
@@ -1,75 +1,100 @@
...
+    image: mysql:5.7
+    environment:
+      MYSQL_ALLOW_EMPTY_PASSWORD: yes
+      MYSQL_DATABASE: test
     when:
       event:
-        exclude: [ pull_request ]
+        - push
+        - tag
+        - pull_request
```
Comment (appleboy) [inline .drone.yml:20]: remove the `when` condition for all services?

Comment (kolaente) [reply .drone.yml:20]: Oh I totally overlooked that, thanks!
Fixed it.

**Comment by appleboy** [comment]
@kolaente LGTM

**Review by lafriks — APPROVED** [review]

**Pull request merged**